torch/csrc/lazy/core/shape.cpp

```diff
@@ -33,5 +33,15 @@ std::vector<Shape> convertShapes(
   return shape;
 }
 
+size_t Shape::numel() const {
+  size_t elts = dim() > 0 ? 1 : 0;
+  for (size_t i = 0; i < dim(); i++) {
+    // TODO(whc) is this check both necessary, and correct?
+    TORCH_CHECK(size(i) > 0);
```

Comment:
Maybe `TORCH_INTERNAL_ASSERT` if you are not sure?

Reply:
I'm confused- Isn't TORCH_INTERNAL_ASSERT a stronger statement of a case that is absolutely a bug if false?  And TORCH_CHECK is somewhat more relaxed?  In any case, i think they do the same thing, but, I am not that sure when to use one or the other.  

Regardless, I'm not sure this check should be here at all.  IT appears that size==0 is actually valid in pytorch, and numel() should return zero in those cases. So i think I need to delete this. 

```
>>> x = torch.ones((1,0,2))
>>> x.numel()
0
```

Reply:
I think the difference is that `TORCH_CHECK` is user facing and `TORCH_INTERNAL_ASSERT` is debug assert. I'm not sure about which one is stronger since they all just crash the program. For error logging, you might want to use `LOG(ERROR)`.